Monika-After-Story/MonikaModDev
# Very Early mas_log

`renpy.log` (or `renpy.exports.log`) is extremely unreliable between init `-999` and `-995`. Sometimes it exists as one thing, sometimes it's not that, additionally because there is no logfile created internally at the time, we simply cannot trust this to log for us at these levels.

As such, I have moved the logging portions of `mas_utils` into `python early`, and it should be the first init block we have to run for MAS.

# Affected Members:
- `MASMacLog`
- `macLogOpen`
- `logcreate`
- `writelog`
- `wtf`
- `writestack`
- `logrotate`
- `tryparseint`
- `mas_mac_log_cache`
- `mas_log`
- `mas_log_open`


# Testing:
- Verify no crashes
- Verify logging still works
- Verify all functions exists as normal
- Verify logs still get rotated
- Verify logs work properly during early init levels (`-999` - `-995`)
